src/match/match.jl

```diff
@@ -24,7 +24,11 @@ isbinding(s) = false
 isbinding(s::Symbol) = occursin(r"[^_]_(_str)?$", string(s))
 
 function bname(s::Symbol)
-  Symbol(Base.match(r"^@?(.*?)_+(_str)?$", string(s)).captures[1])
+  return @static if isdefined(Base, :AbstractMatch)
+    Symbol((Base.match(r"^@?(.*?)_+(_str)?$", string(s))::AbstractMatch).captures[1])
+  else
+    Symbol((Base.match(r"^@?(.*?)_+(_str)?$", string(s))::Base.RegexMatch).captures[1])
+  end
```

Comment:
This change just replaces one runtime error by another, right?

```julia
julia> nothing.captures
ERROR: type Nothing has no field captures

julia> (nothing::AbstractMatch).captures
ERROR: TypeError: in typeassert, expected AbstractMatch, got a value of type Nothing
```

What's the point?

Reply:
AFAIU that runtime error won't happen (the regex always matches `bname`'s argument), thus this annotation.